patches/gcc-10.1.0/0009-static-pie.diff

```diff
@@ -0,0 +1,86 @@
+diff --git a/gcc/common.opt b/gcc/common.opt
+index d33383b523c..49fdd6b42f0 100644
+--- a/gcc/common.opt
++++ b/gcc/common.opt
+@@ -3391,11 +3391,11 @@ Driver
+ 
+ no-pie
+ Driver RejectNegative Negative(shared)
+-Don't create a dynamically linked position independent executable.
++Don't create a position independent executable.
+ 
+ pie
+ Driver RejectNegative Negative(no-pie)
+-Create a dynamically linked position independent executable.
++Create a position independent executable.
+ 
+ static-pie
+ Driver RejectNegative Negative(pie)
+diff --git a/gcc/config/gnu-user.h b/gcc/config/gnu-user.h
+index ff2e880b1fa..448f879d978 100644
+--- a/gcc/config/gnu-user.h
++++ b/gcc/config/gnu-user.h
+@@ -51,13 +51,12 @@ see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
+ #define GNU_USER_TARGET_STARTFILE_SPEC \
+   "%{shared:; \
+      pg|p|profile:%{static-pie:grcrt1.o%s;:gcrt1.o%s}; \
+-     static:crt1.o%s; \
+-     static-pie:rcrt1.o%s; \
++     static|static-pie:%{" PIE_SPEC ":rcrt1.o%s;:crt1.o%s}; \
+      " PIE_SPEC ":Scrt1.o%s; \
+      :crt1.o%s} " \
+    GNU_USER_TARGET_CRTI " \
+-   %{static:crtbeginT.o%s; \
+-     shared|static-pie|" PIE_SPEC ":crtbeginS.o%s; \
++   %{shared|" PIE_SPEC ":crtbeginS.o%s; \
++     static:crtbeginT.o%s; \
+      :crtbegin.o%s} \
+    %{fvtable-verify=none:%s; \
+      fvtable-verify=preinit:vtv_start_preinit.o%s; \
+@@ -76,8 +75,7 @@ see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
+   "%{!static:%{fvtable-verify=none:%s; \
+      fvtable-verify=preinit:vtv_end_preinit.o%s; \
+      fvtable-verify=std:vtv_end.o%s}} \
```

Comment:
According to the [commit](https://gcc.gnu.org/git/?p=gcc.git;a=commit;f=gcc/config/gnu-user.h;h=65a3896a182c3518f3ba7b5cecaf7d1ca7badf0c) that changed this line between GCC 9 and 10, I think we should exclude `static-pie` as well, no?

For example (borrowed from [Szabolcs Nagy](https://port70.net/~nsz/musl/gcc-10.1.0/0004-static-pie.diff)):

```suggestion
   "%{static|static-pie:; \
      fvtable-verify=none:%s; \
      fvtable-verify=preinit:vtv_end_preinit.o%s; \
      fvtable-verify=std:vtv_end.o%s} \
```

Reply:
Good point. Thank you. Seems, you are right I think we should include it, otherwise compiling static-pie binary should give 'multiple definition of ...' linking error. To be honest, I am not sure whether it was a good idea in gcc10 to make tentative variables go into bss .. (I want to be wrong).

Reply:
@richfelker  any thoughts?